### PR TITLE
Add workaround for `Release not found` error

### LINKED ISSUE #26: Issue with release `Release with tag_name 0.xxx.0 not found`
I've recently been getting the following error (it happened in the last 2 releases):

```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.12/x64/bin/autopub", line 8, in <module>
    sys.exit(main())
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/autopub/autopub.py", line 83, in main
    arguments.func(arguments)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/autopub/autopub.py", line 33, in githubrelease
    create_github_release()
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/autopub/create_github_release.py", line 50, in create_github_release
    asset_pattern="dist/*",
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/github_release.py", line 116, in with_check_for_credentials
    return func(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/github_release.py", line 469, in gh_release_create
    gh_asset_upload(repo_name, tag_name, asset_pattern, dry_run=dry_run)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/github_release.py", line 116, in with_check_for_credentials
    return func(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/github_release.py", line 724, in gh_asset_upload
    upload_url = get_release_info(repo_name, tag_name)["upload_url"]
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/github_release.py", line 272, in get_release_info
    raise Exception('Release with tag_name {0} not found'.format(tag_name))
Exception: Release with tag_name 0.100.0 not found
```

See an example here: https://github.com/strawberry-graphql/strawberry/runs/5434280998?check_suite_focus=true

Not sure if something changed in GitHub's API 🤔

### PR DESCRIPTION
**Pull request by patrick91**
Closes #26

### CODE REVIEW

File: autopub/create_github_release.py [hunk 1]
```diff
@@ -47,5 +48,19 @@ def create_github_release():
...
+    # should be returned by the create API.
+    # anyway, this fix might be good enough for the time being
+
+    time.sleep(2)
+
+    gh_asset_upload(
+        REPO_SLUG,
+        tag,
+        pattern="dist/*",
     )
```
Comment (patrick91) [inline autopub/create_github_release.py:66]: we could use something like tenacity here to make it more resilient, but I expect this code to be remove in future

**Review by justinmayer — APPROVED** [review]
Many thanks for the fix, Patrick. Hopefully this will do the trick in the interim! 🤞

**Comment by minchinweb** [comment]
@justinmayer any chance a release containing this could be cut?

**Comment by justinmayer** [comment]
Already tried, but CircleCI failed, as you should be able to see here in this repo.

Rather than figure out why and fix it, at this point I am more inclined to switch to GitHub CI.

**Comment by justinmayer** [comment]
@patrick91: I switched to GitHub Actions, but even with the fix in this PR, I'm still getting a [`Release with tag_name x.x.x not found` error](https://github.com/autopub/autopub/runs/6998604572?check_suite_focus=true#step:5:46). 😢 

_Addendum:_ I realized that this error occurred when AutoPub tried to publish itself, presumably with the previous (unfixed) version. I just manually published v0.2.2 containing this change, so perhaps the problem will be fixed in subsequent deployments.

**Comment by justinmayer** [comment]
After manually publishing AutoPub v0.2.2, I can confirm that at least in one test 👇, the problem seems to have been resolved. Many thanks for the fix, Patrick! 👏 

👆 For reference, the fix was tested successfully via the initial release of the [Pelican TailwindCSS plugin](https://github.com/pelican-plugins/tailwindcss) by @lcfd.

**Comment by patrick91** [comment]
@justinmayer I'm glad it worked!

**Pull request merged**